CRM/ManualDirectDebit/Hook/BuildForm/UpdateSubscription.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * Implements hook on buildForm event of the UpdateSubscription form.
+ */
+class CRM_ManualDirectDebit_Hook_BuildForm_UpdateSubscription {
+
+  /**
+   * Form that is being built.
+   *
+   * @var CRM_Contribute_Form_UpdateSubscription
+   */
+  private $form;
+
+  /**
+   * CRM_ManualDirectDebit_Hook_BuildForm_UpdateSubscription constructor.
+   *
+   * @param \CRM_Contribute_Form_UpdateSubscription $form
+   */
+  public function __construct(CRM_Contribute_Form_UpdateSubscription $form) {
+    $this->form = $form;
+  }
+
+  /**
+   * Implements buildForm hook.
+   */
+  public function buildForm() {
+    $contactID = CRM_Utils_Request::retrieve('cid', 'Positive', $this->form);
```

Comment:
I would move the content of buildForm to two new private methods, one that contains the code which assigns the contact_id and the other one for the code that assigns the selected_mandate_id.